### PR TITLE
Add extra headers in the requests

### PR DESCRIPTION
**Pull request by gtzsec**
I added 2 additional optional HEADERS,

DD_EXTRA_HEADER_1
DD_EXTRA_HEADER_1_VALUE
DD_EXTRA_HEADER_2
DD_EXTRA_HEADER_2_VALUE

 Sometimes Defect Dojo will be behind some sort of WAF or Zero Trust solution, this requires to set some extra authentication tokens before actually reaching Defect Dojo

### CODE REVIEW

**Comment by gtzsec** [comment]
@StefanFl can you please check this out :3

**Comment by StefanFl** [comment]
Sorry @gtzsec, have been quite busy the last weeks. I just merged to other PRs, there are 2 conflicts now. Can you please resolve them? And please add the additional headers in the unit tests.

**Comment by gtzsec** [comment]
@StefanFl ready!

File: dd_import/environment.py [hunk 1]
```diff
@@ -52,32 +56,31 @@ def check_environment_reimport_findings(self):
...
-        print('DD_TEST_NAME:                       ', self.test_name)
-        print('DD_TEST_TYPE_NAME:                  ', self.test_type_name)
-        print('DD_FILE_NAME:                       ', self.file_name)
-        print('DD_ACTIVE:                          ', self.active)
-        print('DD_VERIFIED:                        ', self.verified)
-        print('DD_MINIMUM_SEVERITY:                ', self.minimum_severity)
-        print('DD_GROUP_BY:                        ', self.group_by)
-        print('DD_PUSH_TO_JIRA:                    ', self.push_to_jira)
-        print('DD_CLOSE_OLD_FINDINGS:              ', self.close_old_findings)
-        print('DD_CLOSE_OLD_FINDINGS_PRODUCT_SCOPE:', self.close_old_findings_product_scope)
```
Comment (StefanFl) [inline dd_import/environment.py:70]: This line will be deleted by your changes.

File: dd_import/environment.py [hunk 2]
```diff
@@ -52,32 +56,31 @@ def check_environment_reimport_findings(self):
...
-        print('DD_DO_NOT_REACTIVATE:               ', self.do_not_reactivate)
```
Comment (StefanFl) [inline dd_import/environment.py:71]: This line will be deleted by your changes.

File: dd_import/environment.py [hunk 3]
```diff
@@ -52,32 +56,31 @@ def check_environment_reimport_findings(self):
...
-        print('DD_VERSION:                         ', self.version)
-        print('DD_ENDPOINT_ID:                     ', self.endpoint_id)
-        print('DD_SERVICE:                         ', self.service)
-        print('DD_BUILD_ID:                        ', self.build_id)
-        print('DD_COMMIT_HASH:                     ', self.commit_hash)
-        print('DD_BRANCH_TAG:                      ', self.branch_tag)
-        print('DD_API_SCAN_CONFIGURATION_ID:       ', self.api_scan_configuration_id)
-        print('DD_SOURCE_CODE_MANAGEMENT_URI:      ', self.source_code_management_uri)
```
Comment (StefanFl) [inline dd_import/environment.py:79]: This line will be deleted by your changes.

File: README.md [hunk 4]
```diff
@@ -52,6 +52,38 @@ Please note you have to set the environment variables as described below and mou
 
 All parameters need to be provided as environment variables:
 
+<<<<<<< HEAD
```
Comment (StefanFl) [inline README.md:55]: This is a remain of the conflict and needs to be deleted

File: README.md [hunk 5]
```diff
@@ -52,6 +52,38 @@ Please note you have to set the environment variables as described below and mou
...
+| DD_BUILD_ID           | Optional           | -                | |
+| DD_COMMIT_HASH        | Optional           | -                | |
+| DD_BRANCH_TAG         | Optional           | -                | |
+| DD_API_SCAN_CONFIGURATION_ID | Optional    | -                | Id of the API scan configuration for API based parsers, e.g. SonarQube |
+| DD_SSL_VERIFY         | Optional           | Optional         | Disable SSL verification by setting to `false` or `0`. Default: `true` |
+| DD_EXTRA_HEADER_1         | Optional           | Optional         | If extra header key is needed for auth in wafs or similar |
+| DD_EXTRA_HEADER_1_VALUE   | Optional           | Optional         | The corresponding value for extra header key |
+| DD_EXTRA_HEADER_2         | Optional           | Optional         | If extra header key is needed for auth in wafs or similar |
+| DD_EXTRA_HEADER_2_VALUE   | Optional           | Optional         | The corresponding value for extra header key |
+=======
```
Comment (StefanFl) [inline README.md:86]: This is a remain of the conflict and needs to be deleted

File: README.md [hunk 6]
```diff
@@ -81,6 +113,7 @@ All parameters need to be provided as environment variables:
 | DD_API_SCAN_CONFIGURATION_ID        | Optional           | -                | Id of the API scan configuration for API based parsers, e.g. SonarQube                            |
 | DD_SOURCE_CODE_MANAGEMENT_URI       | Optional           | -                |                                                                                                   |
 | DD_SSL_VERIFY                       | Optional           | Optional         | Disable SSL verification by setting to `false` or `0`. Default: `true`                            |
+>>>>>>> main
```
Comment (StefanFl) [inline README.md:116]: This is a remain of the conflict and needs to be deleted

File: README.md [hunk 7]
```diff
@@ -52,6 +52,38 @@ Please note you have to set the environment variables as described below and mou
...
+| DD_COMMIT_HASH        | Optional           | -                | |
+| DD_BRANCH_TAG         | Optional           | -                | |
+| DD_API_SCAN_CONFIGURATION_ID | Optional    | -                | Id of the API scan configuration for API based parsers, e.g. SonarQube |
+| DD_SSL_VERIFY         | Optional           | Optional         | Disable SSL verification by setting to `false` or `0`. Default: `true` |
+| DD_EXTRA_HEADER_1         | Optional           | Optional         | If extra header key is needed for auth in wafs or similar |
+| DD_EXTRA_HEADER_1_VALUE   | Optional           | Optional         | The corresponding value for extra header key |
+| DD_EXTRA_HEADER_2         | Optional           | Optional         | If extra header key is needed for auth in wafs or similar |
+| DD_EXTRA_HEADER_2_VALUE   | Optional           | Optional         | The corresponding value for extra header key |
+=======
 | Parameter                           | Re-import findings | Import languages | Remark                                                                                            |
```
Comment (StefanFl) [inline README.md:87]: Here the table starts again. The conflict has not been removed properly, now there are the old and the new version of the table in the file.

**Review by StefanFl — CHANGES_REQUESTED** [review]
Unfortunately you haven't resolved the conflict properly, so the table has double entries and some lines are missing in section that prints the environment variables.

**Comment by gtzsec** [comment]
I opened a new MR :)

**Pull request closed without merging**